modules/twitch_streaming

```diff
@@ -0,0 +1,68 @@
+"""
+Checks if a streamer is online. [Gross Gore/grossie_gore as placeholder]
+
+Checks if a streamer is online using the Twitch Kraken API to see
+if a channel is currently streaming or not.
+
+Configuration parameters
+    cache_timeout: how often we refresh this module in seconds
+    stream_name: name of streamer(twitch.tv/<stream_name>)
+
+Format of status string placeholders
+    {stream_name}:  name of the streamer (twitch.tv/<stream_name>)
+
+@author Alex Caswell horatioesf@virginmedia.com
+@license BSD
+"""
+
+# import your useful libs here
+from time import time
+import requests
+
+
+class Py3status:
+    # available configuration parameters
+    # can be customized in i3status.conf
+    cache_timeout = 10
+    stream_name = "grossie_gore"
+    format = "{stream_name} is live!"
+    format_offline = "{stream_name} is offline."
+    format_invalid = "{stream_name} does not exist!"
+
+    def __init__(self):
+        self._display_name = None
+
+    def _get_display_name(self):
+        display_name_request = requests.get('https://api.twitch.tv/kraken/users/' + self.stream_name)
+        self._display_name = display_name_request.json().get('display_name')
+
+    def is_streaming(self, i3s_output_list, i3s_config):
+        r = requests.get('https://api.twitch.tv/kraken/streams/' + self.stream_name)
+        if not self._display_name:
+            self._get_display_name()
+        if 'error' in r.json():
+            colour=i3s_config['color_bad']
+            full_text = self.format_invalid.format(stream_name=self.stream_name)
+        elif r.json().get('stream') == None:
+            colour=i3s_config['color_bad']
+            full_text = self.format_offline.format(stream_name=self._display_name)
+        else:
+            colour=i3s_config['color_good']
+            full_text = self.format.format(stream_name=self._display_name)
```

Comment:
Here you are deciding that if nothing is 'wrong' then everything must be good.  I'd prefer that we check if things are good and if not then assume things are bad.

I haven't checked but I imagine that if we are not on a network that the module will currently show as being online.

Does that make sense?